### ISSUE TITLE
New instance-based GUI wreaks havoc with ThermoPower model parameters

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#10703]
### Description

A single action on the parameters of a ThermoPower array causes a staggering number of different issues.

### Steps to Reproduce

Load the [TestModifiers.mo.txt](https://github.com/OpenModelica/OpenModelica/files/11490329/TestModifiers.mo.txt) model in OMEdit (you need to install ThermoPower, currently listed as no support).

Double-click on the wall component, go to the Initialisation tab, and write 25 in the Tstartbar field, then click OK.

The following error message is displayed:
```
[16] 19:11:37 Translation Error
[ThermoPower.Thermal: 599:5-601:52]: [Non-array modification ‘"K"‘ for array component ‘displayUnit‘, possibly due to missing ‘each‘.](omeditmessagesbrowser:///ThermoPower.Thermal?lineNumber=599)
```
and the diagram disappears because of invalid code. The offending line of code reads:
```modelica
  ThermoPower.Thermal.MetalWallFV wall(M = 1, Nw = 10, cm = 500, Tstartbar = 298.15, 
    Tstart1(displayUnit = "K"), TstartN(displayUnit = "K"), Tvolstart(displayUnit = "K"), Tvol(displayUnit = "K")) 
  annotation(
    Placement(visible = true, transformation(origin = {10, 22}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
```
Now, the Tstartbar modifier is correct (it was converted automatically to K from 25 °C), but the modifier following it are wrong or a number of reasons:
1. "K" is the default display unit for the Temperature type, and that was not changed, so why cluttering the model with displayUnit modifiers in the first place?
2. Tvolstart is an array, so its displayUnit modifier (were it actually changed) should be `(each displayUnit = "K")`, otherwise the front-end complains about wrong types
3. Tvol is a _variable_ of model, not a parameter. Q1: why was a displayUnit added for a variable in the first place? Q2: assuming it made sens to add it, why was it added to the default value of its type? Q3: even in that case, why was it added without `each`, thus making the model invalid, since it is an array?

### Expected Behavior

The only modifier that should be added is Tstartbar = 298.15.

**Comment by casella** [comment]
A closely related issue is #5395, I would suggest to fix that as well.

**Comment by casella** [comment]
Adding @fedetftpolimi to the loop, he actually discovered this issue.

**Comment by adeas31** [comment]
#10741 fixes the issue that it doesn't add unchanged modifiers so the only modifier that should be added is Tstartbar = 298.15.

> 1. "K" is the default display unit for the Temperature type, and that was not change, so why cluttering the model with displayUnit modifiers in the first place?

`K` is the unit and `degC` is the display unit. #10741 fixes this issue that now it shows the correct display unit. If then we change the display unit to `K` then it is added as modifier to the component.

> 2. Tvolstart is an array, so its displayUnit modifier (were it actually changed) should be `(each displayUnit = "K")`, otherwise the front-end complains about wrong types

This is known issue. Lets follow this on #5395. See also #5405, #5737, #7750 and #10226

> 4. Tvol is a _variable_ of model, not a parameter. Q1: why was a displayUnit added for a variable in the first place? Q2: assuming it made sens to add it, why was it added to the default value of its type? Q3: even in that case, why was it added without `each`, thus making the model invalid, since it is an array?

The displayUnit added to `Tvol` will be fixed with #10741. We can add modifiers and display units to variables as well.

**Comment by casella** [comment]
Works fine now, thanks @adeas31!
